packages/@aws-cdk/aws-cognito-identitypool/lib/identitypool-role-attachment.ts

```diff
@@ -0,0 +1,228 @@
+
+  /**
+   * Default Unauthenticated (Guest) Role
+   * @default - No default role will be added
+   */
+  readonly unauthenticated?: IRole;
+}
+
+/**
+ * Map roles to users in the identity pool based on claims from the Identity Provider
+ *  @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-identitypoolroleattachment.html
+ */
+export interface IdentityPoolRoleMapping {
+  /**
+   * The url of the provider of for which the role is mapped
+   */
+  readonly providerUrl: IdentityPoolProviderUrl;
+
+  /**
+   *  If true then mapped roles must be passed through the cognito:roles or cognito:preferred_role claims from identity provider.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/role-based-access-control.html#using-tokens-to-assign-roles-to-users
+   *
+   * @default false
+   */
+  readonly useToken?: boolean;
+
+  /**
+   * Allow for role assumption when results of role mapping are ambiguous
+   * @default false - Ambiguous role resolutions will lead to requester being denied
+   */
+  readonly resolveAmbiguousRoles?: boolean;
+
+  /**
+   * The claim and value that must be matched in order to assume the role. Required if useToken is false
+   * @default - No Rule Mapping Rule
+   */
+  readonly rules?: RoleMappingRule[];
+}
+
+/**
+ * Types of matches allowed for Role Mapping
+ */
+export enum RoleMappingMatchType {
+  /**
+   * The Claim from the token must equal the given value in order for a match
+   */
+  EQUALS = 'Equals',
+
+  /**
+   * The Claim from the token must contain the given value in order for a match
+   */
+  CONTAINS = 'Contains',
+
+  /**
+   * The Claim from the token must start with the given value in order for a match
+   */
+  STARTS_WITH = 'StartsWith',
+
+  /**
+   * The Claim from the token must not equal the given value in order for a match
+   */
+  NOTEQUAL = 'NotEqual',
+}
+
+/**
+ * Represents an Identity Pool Role Attachment Role Mapping Rule
+ */
+export interface RoleMappingRule {
+  /**
+   * The key sent in the token by the federated identity provider.
+   */
+  readonly claim: string;
+
+  /**
+   * The Role to be assumed when Claim Value is matched.
+   */
+  readonly mappedRole: IRole;
+
+  /**
+   * The value of the claim that must be matched
+   */
+  readonly claimValue: string;
+
+  /**
+   * How to match with the Claim value
+   * @default RoleMappingMatchType.EQUALS
+  */
+  readonly matchType?: RoleMappingMatchType
+}
+
+/**
+ * Defines an Identity Pool Role Attachment
+ *
+ * @resource AWS::Cognito::IdentityPoolRoleAttachment
+ */
+export class IdentityPoolRoleAttachment extends Resource implements IIdentityPoolRoleAttachment {
+  /**
+   * Id of the underlying identity pool
+   */
+  public readonly identityPoolId: string
+
+  constructor(scope: Construct, id: string, props: IdentityPoolRoleAttachmentProps) {
+    super(scope, id, {
+      physicalName: props.identityPoolRoleAttachmentName || Lazy.string({
```

Comment:
I don't think `physicalName` is needed for this one. It doesn't look like you can provide a `name` for the `CfnIdentityPoolRoleAttachment`

Reply:
I did this because having `<Construct>Name` as a part of the Props interface appears to be standard, and without this that prop would serve no purpose.

Reply:
Did you get any error when you didn't provide the name property? I definitely think we should not provide the name since it doesn't get used anywhere.